frontend/src/components/customers/CustomerReservation.js

```diff
@@ -108,59 +59,138 @@ function CustomerReservation(props) {
     },
     [stageSelectionChanged]
   );
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    let date = new Date().toISOString();
+    date = date.substr(0, date.indexOf("T"));
+
+    axios
+      .post(
+        `${props.backendDomain}reservation`,
+        {
+          date: date,
+          startTime: timeSlots[timeSlotId].startTime,
+          endTime: timeSlots[timeSlotId].endTime,
+          customer: props.customerId,
+          stylist: stylistId,
+          service: serviceIds,
+        },
+        {
+          headers: { Authorization: `JWT ${localStorage.getItem("token")}` },
+        }
+      )
+      .then(() => {
+        setSubmitted(true);
+      })
+      .catch((err) => {
+        window.alert("An error has occurred. Please try again.");
+        console.log(err);
+        history.push("/customers/home");
+      });
+  };
+
+  //TODO: Proper error handling
 
   // Get services
 
   useEffect(() => {
```

Comment:
I think it would be beneficial for readability if all "useEffect" are at the top, after the definition of the state variables.

Reply:
Done (except for the one function that needs to be declared before the effects)